tests/tests.py

```diff
@@ -1,7 +1,31 @@
 import warnings
 
-from django.template import engines
-from django.test import TestCase
+import django.template
+from django.test import TestCase, override_settings
+from django.template import engines, EngineHandler
+
+from template_partials.apps import wrap_loaders
+
+
+@override_settings(
+    TEMPLATES=[
+        {
+            "BACKEND": "django.template.backends.django.DjangoTemplates",
+            "APP_DIRS": True,
+        },
+    ]
+)
+class SimpleAppConfigTestCase(TestCase):
+    def test_wrap_loaders(self):
+        django.template.engines = EngineHandler()
```

Comment:
Hi @jnns thanks for this. 

It looks right. Could you just though point to what it is in slippers that requires this? (Why are they instantiating the engine at app registration point, I wonder? 🤔)

Reply:
Slippers uses a `components.yml` file in which all component names and their template files are to be registered. 

This file is loaded at app configuration time using `django.template.loader.select_template`. When `select_template` is imported, the loaders are configured and cached. 

Subsequent apps can change the loader settings but that doesn't change the cache.

Reply:
Don't suppose you could drop a link to that to save me a cycle could you? 

I want to confirm that clearing the cache is the correct response, rather than say raising a configuration error...

Thanks!

Reply:
Sure thing! This is a (simplified) call stack of what happens in Slipper's AppConfig:

- slippers/apps.py ready()  
    register_tags()
  -  slippers/apps.py register_tags()
     template = get_components_yaml() 
     - [slippers/apps.py get_components_yaml()](https://github.com/mixxorz/slippers/blob/53d2ac670949114ece8ba8622f8238a842a041ff/slippers/apps.py#L14)
       return select_template(["components.yaml", "components.yml"])
       - [django/template/loader.py select_template()](https://github.com/django/django/blob/61c305f298da1b4079a80721c861d0663dc8717e/django/template/loader.py#L22)
         engines = _engine_list(using)
         - [django/template/utils.py EngineHandler](https://github.com/django/django/blob/61c305f298da1b4079a80721c861d0663dc8717e/django/template/utils.py#L26)  
           
When the engine list is populated, the EngineHandler gets instantiated with `settings.TEMPLATES` and caches it:

```python
class EngineHandler:
    @cached_property
    def templates(self):
        self._templates = settings.TEMPLATES
```

Raising a ConfigurationError would also be an option, I think. But in that case we must be sure that the installation instructions we give our users are future proof. Meaning: `template_partials` must be the first app in `INSTALLED_APPS` immediately after the django core packages (or the very first even, if we can't be sure that a django.contrib package imports `django.template.loader` during startup).

Reply:
Yep, great thanks. I can't see any reason why we can't clear the cache here. 👍 

Thanks again 🎁